misc/metrics/src/bandwidth.rs

```diff
@@ -0,0 +1,312 @@
+use crate::protocol_stack;
+use futures::{
+    future::{MapOk, TryFutureExt},
+    io::{IoSlice, IoSliceMut},
+    prelude::*,
+    ready,
+};
+use libp2p_core::{
+    muxing::{StreamMuxer, StreamMuxerEvent},
+    transport::{ListenerId, TransportError, TransportEvent},
+    Multiaddr,
+};
+use libp2p_identity::PeerId;
+use prometheus_client::{
+    encoding::{EncodeLabelSet, EncodeLabelValue},
+    metrics::{counter::Counter, family::Family},
+    registry::{Registry, Unit},
+};
+use std::{
+    convert::TryFrom as _,
+    io,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+#[derive(Debug, Clone)]
+#[pin_project::pin_project]
+pub struct Transport<T> {
+    #[pin]
+    transport: T,
+    metrics: Family<Labels, Counter>,
+}
+
+impl<T> Transport<T> {
+    pub fn new(transport: T, registry: &mut Registry) -> Self {
+        let metrics = Family::<Labels, Counter>::default();
+        registry
+            .sub_registry_with_prefix("libp2p")
```

Comment:
I would expect the user to already pass us a registry that is scoped to `libp2p`.

Reply:
Oh yeah. Should definitrly stay consitent!

Fine as is.